[MIG] account_operating_unit: Migration to 17.0

jdidderen-nsi opened the pull request:
Depends on:

- [x] https://github.com/OCA/operating-unit/pull/667
- [x] https://github.com/OCA/operating-unit/pull/669

sudhir-serpentcs committed:
[ADD] Added Account Operating Unit module for v9.

JordiBForgeFlow committed:
[FIX] various fixes to ensure that unit tests pass

JordiBForgeFlow committed:
[IMP] changed Eficent copyright

serpentcs-dev1 committed:
[IMP] : Migrated the code to version 9

serpentcs-dev1 committed:
[IMP] Migrated and tested account_ou module to v9.

JordiBForgeFlow committed:
[FIX] travis errors

JordiBForgeFlow committed:
[FIX] remove period_id

JordiBForgeFlow committed:
[FIX] Applied review comments by @elicoidal

JordiBForgeFlow committed:
[FIX] errors in creation of account move

JordiBForgeFlow committed:
[FIX] README.RST style

JordiBForgeFlow committed:
[FIX] travis errors

JordiBForgeFlow committed:
[FIX] travis

JordiBForgeFlow committed:
[IMP] add invoice analysis by operating unit

AaronHForgeFlow committed:
[FIX] Do not assert_balance until all journal items exist. Fix search view in report. Other minor fixes.

AaronHForgeFlow committed:
[FIX] Style fixes suggested by @elicoidal

AaronHForgeFlow committed:
[FIX] travis issues

AaronHForgeFlow committed:
[FIX] travis issues

AaronHForgeFlow committed:
[UPT] By default the OU is not self balanced because the constraint of setting an inter-OU accounts triggers  from other modules when creating companies

AaronHForgeFlow committed:
[UPT] suggested type for inter OU account

JordiBForgeFlow committed:
[FIX] improve README. Inter OU clearing is a balance sheet account

JordiBForgeFlow committed:
[IMP] Add the operating units in the financial reports qweb

AaronHForgeFlow committed:
[FIX] travis issues

JordiBForgeFlow committed:
[IMP] various additions, including:
* Add management of payments using operating units
* Trial Balance by operating units
* Improve tests

JordiBForgeFlow committed:
[IMP] simplify headers

JordiBForgeFlow committed:
[IMP] update readme

JordiBForgeFlow committed:
[FIX] PEP8 issues

JordiBForgeFlow committed:
[IMP] add security for account journals

JordiBForgeFlow committed:
[IMP] add security by OU on the account invoice report

JordiBForgeFlow committed:
[FIX] flake8

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

LoisRForgeFlow committed:
[MIG] account_operating_unit to v10.0

AaronHForgeFlow committed:
[FIX] Company view

AaronHForgeFlow committed:
[FIX] Test remove workflow

JordiBForgeFlow committed:
[10.0][account_operating_unit]Add support for cash basis. Migrate to v10 nomenclature (#55)

* Add support for cash basis. Migrate to v10 nomenclature

* Remove apply_taxes from the create method, as in v10 this parameter is not accepted in the created method

JordiBForgeFlow committed:
Add dependency on analytic_operating_unit. (#62)

The analytic account to be selected in customer
or supplier invoices must have the operating unit
of the invoice as an allowed operating unit.

LoisRForgeFlow committed:
[FIX] fix wrong iteration.

AaronHForgeFlow committed:
[FIX]When paying, the OU of the invoice should be considered only when a single invoice is being paid

serpentcs-dev1 committed:
[FIX] readonly on OU

Commit:
[FIX][account_payment]Fix in the internal transfers amount currency

sbidoul committed:
Allow selecting OU on all account journals

Closes #118

AaronHForgeFlow committed:
[FIX]PYLINT mainly

Commit:
[MIG]account_operating_unit to v11

Commit:
Remove erroneous extra currency code.

To my understanding - the currency lines in liquidity (_get_dst_liquidity_aml_dict_vals) are just wrong and should not be here at all. I understand the need to refactor to get operating unit in, but base Odoo does not have this code about currency.
Now, this fails if the currency and the company currency do not match the journal / target currency.
E.g. A company in Australian Currency transfers 100 Euro from a Euro account to a USD account. When writing to the USD side, self.currency_id will be Euro and self.company_id.currency_id will be AUD, but the transaction will be expecting the USD equivalent.
Base Odoo is currenly handling it all correctly in '_get_shared_move_line_vals', but this code breaks it.

grindtildeath committed:
Changes according to reviews on #114

grindtildeath committed:
Fix default operating_unit_id on account.move.line

bjeficent committed:
[MIG] account_operating_unit: Migration to v12

bjeficent committed:
[FIX] Counterpart's operating unit is wrong and Journals filtered by OUs

jose-pcg committed:
[FIX] account_operating_unit: wrong account.move.line method name calling

oca-travis committed:
[UPD] Update account_operating_unit.pot

OCA-git-bot committed:
[UPD] README.rst

MiquelRForgeFlow committed:
[FIX] Alignments

oca-travis committed:
[UPD] Update account_operating_unit.pot

nikul-serpentcs committed:
[FIX] Fixed unknown Fields

OCA-git-bot committed:
account_operating_unit 12.0.1.0.1

osi-scampbell committed:
[IMP] Ensure Company Set

OCA-git-bot committed:
account_operating_unit 12.0.1.0.2

alan196 committed:
[IMP] account_operating_unit: black, isort, prettier

alan196 committed:
[MIG] account_operating_unit: Migration to V13

oca-travis committed:
[UPD] Update account_operating_unit.pot

newtratip committed:
[13.0][IMP] Add some feature in account_operating_unit module

newtratip committed:
[13.0][IMP] Adjust constrains _check_journal_operating_unit

oca-travis committed:
[UPD] Update account_operating_unit.pot

OCA-git-bot committed:
account_operating_unit 13.0.1.1.0

Commit:
Reduce the width of dependency from analytic_operating_unit to operating_unit

OCA-git-bot committed:
account_operating_unit 13.0.1.2.0

oca-travis committed:
[UPD] Update account_operating_unit.pot

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

OCA-git-bot committed:
[UPD] README.rst

alan196 committed:
[IMP] account_operating_unit: black, isort, prettier

alan196 committed:
[MIG] account_operating_unit: Migration to 14.0

oca-travis committed:
[UPD] Update account_operating_unit.pot

kittiu committed:
[14.0][FIX] account_operating_unit, fix test require bill date

alan196 committed:
[IMP] account_operating_unit: black, isort, prettier

alan196 committed:
[MIG] account_operating_unit: Migration to 15.0

Saran440 committed:
[FIX] domain field ou and test script

ps-tubtim committed:
[FIX] account_operating_unit: add OU to cash basis journal entry

ps-tubtim committed:
[FIX] account_operating_unit: raise error when group payment from moves with difference OUs

Commit:
[UPD] Update account_operating_unit.pot

OCA-git-bot committed:
[UPD] README.rst

hieulucky111 committed:
[15.0][FIX]-account_operating_unit: fix the error 'You are trying to reconcile some entries that are already reconciled'

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_operating_unit 15.0.1.0.1

Saran440 committed:
[FIX] account_operating_unit: re reconcile payment for case self balance only

OCA-git-bot committed:
account_operating_unit 15.0.1.0.2

AaronHForgeFlow committed:
[FIX]account_operating_unit: Ensure active_model is account.move before attempting to browse

OCA-git-bot committed:
account_operating_unit 15.0.1.0.3

ps-tubtim committed:
[FIX] account_operating_unit: add OU to tree view of account.move.line

OCA-git-bot committed:
account_operating_unit 15.0.1.1.0

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 100.0% (26 of 26 strings)

Translation: operating-unit-15.0/operating-unit-15.0-account_operating_unit
Translate-URL: https://translation.odoo-community.org/projects/operating-unit-15-0/operating-unit-15-0-account_operating_unit/es_MX/

Saran440 committed:
[FIX] account_operating_unit: split function support expense operating unit

Saran440 committed:
[IMP] account_operating_unit: add OU in move line outstanding payment

OCA-git-bot committed:
account_operating_unit 15.0.1.1.1

OCA-git-bot committed:
account_operating_unit 15.0.1.1.2

AaronHForgeFlow committed:
[FIX] account_operating_unit: assign operating unit of the invoice if journal has no operating unit no matter if OU are self balanced

Saran440 committed:
[FIX] account_operating_unit: domain analytic from operating unit line

OCA-git-bot committed:
account_operating_unit 15.0.1.2.0

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
oca-github-bot post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: operating-unit-15.0/operating-unit-15.0-account_operating_unit
Translate-URL: https://translation.odoo-community.org/projects/operating-unit-15-0/operating-unit-15-0-account_operating_unit/

robyf70 committed:
[15.0][FIX] Fix operating unit payment tests

AaronHForgeFlow committed:
[IMP] account_operating_unit: allow to pay multiple invoices
with different operating units.

That was possible in old versions <v12 but it was removed for unknown
reason.

In general, reconciling a bank journal entry with an invoice
will require Inter OU balancing entries, as long one OU is paying
the balance for the other.

Commit:
[UPD] Update account_operating_unit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: operating-unit-15.0/operating-unit-15.0-account_operating_unit
Translate-URL: https://translation.odoo-community.org/projects/operating-unit-15-0/operating-unit-15-0-account_operating_unit/

Commit:
[IMP] account_operating_unit: pre-commit execution

Commit:
[MIG] account_operating_unit: Migration to 17.0

jdidderen-nsi commented:
As soon as #667 is merged, everything should be good to go for this one.

yankinmax commented on account_operating_unit/models/account_move.py:
```diff
@@ -22,16 +22,6 @@ def create(self, vals_list):
                     vals["operating_unit_id"] = move.operating_unit_id.id
         return super().create(vals_list)
 
-    @api.model
```
why is is just removed?

jdidderen-nsi replied:
It was removed from OC in v16 (https://github.com/odoo/odoo/pull/96134) and it seems that it's not used anymore outside of reports like account.invoice.report

yankinmax commented on account_operating_unit/models/account_move.py:
```diff
@@ -170,16 +153,13 @@ def _onchange_operating_unit(self):
+    def _compute_operating_unit(self):
+        for record in self:
+            if record.journal_id.operating_unit_id:
+                record.operating_unit_id = record.journal_id.operating_unit_id.id
```
you don't need to explicitly assign an `id` to a many2one record, just `record.journal_id.operating_unit_id`

jdidderen-nsi replied:
Updated. Thanks for the review 👍

yankinmax commented on account_operating_unit/report/account_invoice_report.py:
```diff
@@ -19,10 +19,3 @@ def _select(self):
         """
         return select_str
-
-    def _group_by(self):
```
why is it just removed?

jdidderen-nsi replied:
It was removed from OC in v16 (https://github.com/odoo/odoo/pull/96134) and it seems that it's not used anymore outside of reports like account.invoice.report

yankinmax commented on account_operating_unit/views/account_move_view.xml:
```diff
@@ -79,50 +64,32 @@
-            </xpath>
             <field name="line_ids" position="attributes">
-                <attribute name="context">
+                <attribute name="context" operation="update">
```
`operation="update"` will be only applied if you have a dependency on `base_view_inheritance_extension` I suppose, so probably you want to overwrite the original context like it was before

jdidderen-nsi replied:
Yes it's exactly that

yankinmax commented:
> As soon as #667 and #669 is merged, everything should be good to go for this one.

why don't simply add `tets-requirements.txt` file including dependencies references?
just name that commit with prefix `DON'T MERGE`

ghost commented:
> > As soon as #667 and #669 is merged, everything should be good to go for this one.
> 
> why don't simply add `tets-requirements.txt` file including dependencies references? just name that commit with prefix `DON'T MERGE`

Thanks for the tip :)

Camille0907 commented on account_operating_unit/tests/test_payment_operating_unit.py:
```diff
@@ -76,7 +76,7 @@ def test_payment_from_two_invoices(self):
             # Validate that inter OU balance move lines are created
             self.assertEqual(len(payment.move_id.line_ids), 2)
-            self.assertAlmostEqual(payment.amount, 115000)
+            self.assertEqual(payment.amount, invoices[0].amount_total)
```
No risk of float precision issue ?

jdidderen-nsi replied:
I don't think so. Since the decimal precision of the amount field is based on the currency, we should not have any difference between the amount of the invoice and the amount of the payment.

Camille0907 commented on account_operating_unit/models/account_move.py:
```diff
@@ -241,19 +211,16 @@ def _post(self, soft=True):
+                amls = ml_obj.with_context(check_move_validity=False).create(line_datas)
             if amls:
-                move.with_context(wip=False).write(
+                move.with_context(check_move_validity=True).write(
```
Is `check_move_validity` a new key you just created or already existing somewhere ?
Asking because if it's a new one, it isn't read anywhere, so I don't understand its purpose

jdidderen-nsi replied:
It's a key added in Odoo Community a few version ago (https://github.com/odoo/odoo/blob/96223d322c0f59b35ccda01c82e99a1ac6e2fe0c/addons/account/models/account_move.py#L1890). 
The purpose is to avoid the check on the balance.

Camille0907 replied:
Alright; thank you :)

Camille0907 reviewed (approved):
Migration technically LGTM

alan196 committed:
[FIX] account_operating_unit: add operating unit to cash basis counterpart base line

This commit adds the operating unit to the cash basis counterpart base line
when cash basis accounting is enabled.

yankinmax commented:
@jdidderen-nsi can you please remove dependency commit and rebase?
operating_unit is merged:
- [x] https://github.com/OCA/operating-unit/pull/667

ghost commented:
@yankinmax And it's rebased 👍

hbrunn commented:
I was curious how you solved adding the operating unit to the analytic distribution widget, as I mentioned in the [v16 migration](https://github.com/OCA/operating-unit/pull/656#issue-2186979149), but it seems you didn't, correct? In my opinion this needs to be done for the migration to be complete

ghost commented:
@hbrunn It was solved directly in the analytic_operating_unit module - https://github.com/OCA/operating-unit/pull/669

hbrunn commented:
ah nice, thanks. That would then also be a good solution for v16, will backport that eventually

hbrunn reviewed (approved):
please also cherry pick https://github.com/OCA/operating-unit/pull/685 so that we have feature parity

hbrunn committed:
[ADD] account_operating_unit: increase test coverage

hbrunn commented:
please also add https://github.com/OCA/operating-unit/pull/656/commits/d328870c717f758f1af0eab7e3da6de118c9d602 for higher test coverage

yankinmax commented:
@jdidderen-nsi can you pls apply @hbrunn last suggestions?

ghost commented:
> please also cherry pick #685 so that we have feature parity

> please also add [d328870](https://github.com/OCA/operating-unit/commit/d328870c717f758f1af0eab7e3da6de118c9d602) for higher test coverage

> @jdidderen-nsi can you pls apply @hbrunn last suggestions?

@hbrunn @yankinmax And it's done 👍

hbrunn commented:
/ocabot merge nobump

AaronHForgeFlow commented:
Issue with pre-commit on build, trying it again

/ocabot merge nobump
